apps/studio/src/components/ResourceSelector/ResourceSelectorContent.tsx

```diff
@@ -0,0 +1,150 @@
+import { Text, VStack } from "@chakra-ui/react"
+import { Button } from "@opengovsg/design-system-react"
+
+import type { ResourceItemContent } from "~/schemas/resource"
+import { ResourceItem, ResourceItemSkeleton } from "./ResourceItem"
+import { lastResourceItemInAncestryStack } from "./utils"
+
+const NoItemsInFolderResult = () => {
+  return (
+    <Text
+      h="full"
+      w="full"
+      textAlign="center"
+      textStyle="caption-2"
+      display="flex"
+      alignItems="center"
+      justifyContent="center"
+    >
+      This folder is empty.
+    </Text>
+  )
+}
+
+const ResourceItemsResults = ({
+  resourceItemsWithAncestryStack,
+  isResourceIdHighlighted,
+  isResourceItemDisabled,
+  hasAdditionalLeftPadding,
+  handleClickResourceItem,
+}: {
+  resourceItemsWithAncestryStack: ResourceItemContent[][]
+  isResourceIdHighlighted: (resourceId: string) => boolean
+  isResourceItemDisabled: (resourceItem: ResourceItemContent) => boolean
+  hasAdditionalLeftPadding: boolean
+  handleClickResourceItem: (
+    resourceItemWithAncestryStack: ResourceItemContent[],
+  ) => void
```

Comment:
style: usually if it gets so big the props shud be typed outside but not really a big issue

Reply:
addressed in https://github.com/opengovsg/isomer/pull/865/commits/0f3333eda8216a3d3ef229592e13cf04dc31aef3